crates/protocol/src/compression/variant.rs

```diff
@@ -1,15 +1,17 @@
-//! A variant over the different implementations of [Compressor].
+//! A variant over the different implementations of [ChannelCompressor].
 
-use crate::{BrotliCompressor, CompressionAlgo, Compressor, ZlibCompressor};
-use alloc::vec::Vec;
+use crate::{
+    BrotliCompressor, ChannelCompressor, CompressionAlgo, CompressorResult, CompressorWriter,
+    ZlibCompressor,
+};
 use op_alloy_genesis::RollupConfig;
 
 /// The channel compressor wraps the brotli and zlib compressor types,
-/// implementing the [Compressor] trait itself.
+/// implementing the [ChannelCompressor] trait itself.
 #[derive(Debug, Clone)]
 pub enum VariantCompressor {
     /// The brotli compressor.
-    Brotli(BrotliCompressor),
+    Brotli(Box<BrotliCompressor>),
     /// The zlib compressor.
     Zlib(ZlibCompressor),
 }
```

Comment:
compressor trait looks object safe, so I assume `Box<dyn Compressor>` should also work

Reply:
The `Box` wrapper was legacy due to the streaming compression writer impl. That will be a follow-up pr